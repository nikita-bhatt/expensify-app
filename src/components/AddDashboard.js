import React from 'react';
const AddDashboardPage = () => (
    <div>
        This is Add dashboard component
    </div>

);

export default AddDashboardPage;