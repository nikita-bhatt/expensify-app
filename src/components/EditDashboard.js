import React from 'react';
const EditDashboardPage = () => (
    <div>
        This is Edit dashboard component
    </div>

);
export default EditDashboardPage;