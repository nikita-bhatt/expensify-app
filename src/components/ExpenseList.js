import React from 'react';
import { connect } from "react-redux";
import filters from '../reducers/filters';

const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {props.expenses.length}
        {props.filters.text}
    </div>
);
//map store state to component prop
const mapStateToProps = (state) => {
    return {
        expenses: state.expenses,
        filters: state.filters

    };

}
export default connect(mapStateToProps)(ExpenseList);;