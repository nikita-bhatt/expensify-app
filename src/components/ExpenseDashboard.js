import React from 'react';
import ExpenseList from './ExpenseList';
import ExenseListItem from './ExpenseListItem'
import ExpenseListItem from './ExpenseListItem';
const ExpenseDashboardPage = () => (
    <div>
        <ExpenseList/>
    </div>

);
export default ExpenseDashboardPage;