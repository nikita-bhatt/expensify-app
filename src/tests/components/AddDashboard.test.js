import React from 'react';
import {shallow} from 'enzyme';
import {AddDashboardPage} from '../../components/AddDashboard';
import expenses from '../fixtures/expenses';

let onSubmit, history, wrapper;
beforeEach(() => {
    onSubmit = jest.fn(); //spy
    history = {push: jest.fn()};
    wrapper = shallow(<AddDashboardPage onSubmit= {onSubmit} history= {history}/>);

})

test('should render Add Expense dashboard page correctly', () => {  
    expect(wrapper).toMatchSnapshot();
});

test('should handle on Submit', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(onSubmit).toHaveBeenLastCalledWith(expenses[1]);
});