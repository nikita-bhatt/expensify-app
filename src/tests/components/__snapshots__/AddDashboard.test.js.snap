// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Add Expense dashboard page correctly 1`] = `
<div>
  Add Expense
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
